src/main.rs

```
@@ -606,14 +642,28 @@ impl NodeHooks<TransactionLogData> for CanonicalAddressCache {
             .get_block_special_events(finalized_block_info.height)
             .await?
             .response
+            .try_filter_map(|upward| {
+                async move {
+                    match upward {
+                        Upward::Known(special_transaction_outcome) => {
+                            Ok(Some(special_transaction_outcome))
+                        }
+                        Upward::Unknown => Ok(None), // we ignore unknown special events, treat this as no special ones
+                    }
+                }
+            })
             .try_collect()
             .await?;
 
         // Map account addresses affected by each summary to their respective canonical
         // address
         let mut with_addresses = Vec::with_capacity(transaction_summaries.len());
         for summary in transaction_summaries {
-            let affected_addresses = summary.affected_addresses();
+            let affected_addresses = match summary.affected_addresses() {
+                Upward::Known(addresses) => addresses,
+                Upward::Unknown => Vec::new(), // returning empty vec if unknown, the next line will be processing zero length vec anyway, so nothing happens
+            };
+
```

This is not correct behaviour, you will ignore transaction types that are unknown. You can project to `account_transaction` again here before calling `affected_addresses`, but you will need to fail when that function returns `Unknown`

---

throwing an error seems missing.